src/Services/InsuranceService.php

```diff
@@ -60,25 +79,116 @@ public function validate($data)
         return $validator->validate($data);
     }
 
-    public function getOne($pid, $type)
+    public function getOne($uuid, $type = 'primary')
     {
-        $sql = "SELECT * FROM insurance_data WHERE pid=? AND type=?";
+        //Handling when pid and type is sent
+        if (gettype($uuid) == "integer") {
+            $sql = "SELECT * FROM insurance_data WHERE pid=? AND type=?";
+            return sqlQuery($sql, array($uuid, $type));
+        }
 
-        return sqlQuery($sql, array($pid, $type));
+        $processingResult = new ProcessingResult();
+        $isValid = $this->coverageValidator->validateId('uuid', self::COVERAGE_TABLE, $uuid, true);
+        if ($isValid !== true) {
+            return $isValid;
+        }
+        $uuidBytes = UuidRegistry::uuidToBytes($uuid);
+        $sql = "SELECT * FROM insurance_data WHERE uuid=? ";
+
+        $sqlResult = sqlQuery($sql, array($uuidBytes));
+        if ($sqlResult) {
+            $sqlResult['uuid'] = UuidRegistry::uuidToString($sqlResult['uuid']);
+            $processingResult->addData($sqlResult);
+        } else {
+            $processingResult->addInternalError("error processing SQL");
+        }
+        return $processingResult;
     }
 
-    public function getAll($pid)
+    public function getAll($search = array(), $isAndCondition = true)
```

Comment:
note this function is not used in core codebase, so it does not need to support pid

Reply:
Done